src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -156,6 +167,129 @@ private VariantOverlapAnnotator initializeOverlapAnnotator(final FeatureInput<Va
+            if (annotationType instanceof ReducibleAnnotation) {
+
+                ReducibleAnnotation currentASannotation = (ReducibleAnnotation) annotationType;
+
+                final Map<String, Object> annotationsFromCurrentType = currentASannotation.finalizeRawData(vc, originalVC);
+                if (annotationsFromCurrentType != null) {
+                    VariantAnnotations.putAll(annotationsFromCurrentType);
+                }
+                //clean up raw annotation data after annotations are finalized
+                VariantAnnotations.remove(currentASannotation.getRawKeyName());
+            }
+        }
+
+        // generate a new annotated VC
+        final VariantContextBuilder builder = new VariantContextBuilder(vc).attributes(VariantAnnotations);
+
+        // annotate genotypes, creating another new VC in the process
+        final VariantContext annotated = builder.make();
+        return annotated;
+    }
+
+    /**
+     * Calculate the final annotation value from the raw data
+     * @param vc -- contains the final set of alleles, possibly subset by GenotypeGVCFs
+     * @return
+     */
+    public VariantContext annotateContextForActiveRegion(final ReferenceContext referenceContext,
+                                                         final ReadLikelihoods<Allele> likelihoods,
+                                                         final VariantContext vc,
+                                                         final boolean useRaw) {
+        // annotate genotypes
+        final VariantContextBuilder builder = new VariantContextBuilder(vc).genotypes(annotateGenotypes(referenceContext, vc, likelihoods, (s) -> useRaw));
+        VariantContext newGenotypeAnnotatedVC = builder.make();
+
+        final Map<String, Object> infoAnnotations = new LinkedHashMap<>(newGenotypeAnnotatedVC.getAttributes());
+
+        // go through all the requested info annotationTypes that are reducible
+        if (useRaw) {
+            for (final InfoFieldAnnotation annotationType : this.infoAnnotations) {
+                if (!(annotationType instanceof ReducibleAnnotation)) {
+                    continue;
+                }
+
+                ReducibleAnnotation currentASannotation = (ReducibleAnnotation) annotationType;
+                final Map<String, Object> annotationsFromCurrentType = currentASannotation.annotateRawData( referenceContext, newGenotypeAnnotatedVC, likelihoods);
+                if (annotationsFromCurrentType != null) {
+                    infoAnnotations.putAll(annotationsFromCurrentType);
+                }
+            }
+        }
+        //if not in reference-confidence mode, do annotate with reducible annotations, but skip the raw data and go straight to the finalized values
+        else {
+            for (final InfoFieldAnnotation annotationType : this.infoAnnotations) {
+                if (!(annotationType instanceof ReducibleAnnotation)) {
```

Comment:
There is a strong overlap between the reducible annotations and the ActiveRegionBasedAnnotations in GATK3, though its not exactly 1:1 upon inspection. It is using the AcitveRegionBasedAnnotation as a filter then performing the same reduction here. It seems that in GATK3 every reducible annotation falls under this flag, which means that check was to filter the non-reducible annotations, but they couldn't be in the RequestedReducibleAnnotationMap anyway so this might be a spurious check. Since this is for a method that I will probably remove upon completion of the HaplotypeCaller integration of this code I will leave it for more close analysis then.